### PR TITLE
implement data store

### LINKED ISSUE #3: Need interface to data store.
The data store is for persisting timers in a way that they can be queried for "get timer" requests.

### PR DESCRIPTION
**Pull request by nivista**
closes #3

### CODE REVIEW

File: store/client.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+const (
+	http     taskType     = "HTTP"
+	cron     scheduleType = "CRON"
+	interval scheduleType = "INTERVAL"
+)
+
+// NewClient returns a new client to the data store.
+func NewClient(ctx context.Context) (*Client, error) {
+	dbpool, err := pgxpool.Connect(context.Background(), os.Getenv("DATABASE_CONN"))
```
Comment (andrewjdawson2016) [inline store/client.go:33]: That environment variable should be loaded from a configuration file. We can punt on this for now but when your app starts up it should read some config file to know how it should boot up.

File: store/client.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &Client{dbpool}, nil
+}
+
+// Close closes the database client.
+func (c *Client) Close() {
+	c.conn.Close()
```
Comment (andrewjdawson2016) [inline store/client.go:42]: Does this return an error?

File: store/client.go (hunk 1 above)
Comment (andrewjdawson2016) [inline store/client.go:33]: Personally I don't like doing anything in a constructor. I prefer for connection objects like this to have a start and a stop method method.

File: store/client.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	if err != nil {
+		return err
+	}
+	sched, err := json.Marshal(t.Schedule)
+	if err != nil {
+		return err
+	}
+
+	_, err = c.conn.Query(ctx,
+		`insert into timers (id, domain, executionCount, taskType, task, scheduleType, schedule, meta)
```
Comment (andrewjdawson2016) [inline store/client.go:75]: queries should be factored out into some constants.

File: store/client.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+	}
+
+	switch st {
+	case cron:
+		var schedule timer.Cron
+		err = json.Unmarshal(scheduleBytes, &schedule)
+		if err != nil {
+			return nil, err
+		}
+		t.Schedule = &schedule
```
Comment (andrewjdawson2016) [inline store/client.go:127]: ya we cannot do this marshalling of interfaces its weird -we need some serializable database types.

**Review by andrewjdawson2016 — CHANGES_REQUESTED** [review]
- Need serializable types for database 
- Please check in schema definition file its useful for readers of your code. 
- How do you plan to test your application if you don't have a mock for your persistence interface? Anything that depends on it needs a mock.

File: store/client.go (hunk 4 above)
Comment (nivista) [reply store/client.go:127]: Ok, I feel this is a similar situation as on the timer model, lets resolve that and then i'll understand how to do this better.

File: store/client.go (hunk 2 above)
Comment (nivista) [reply store/client.go:42]: Close() doesn't return an error.

File: store/client.go (hunk 1 above)
Comment (nivista) [reply store/client.go:33]: Ok

File: store/client.go (hunk 3 above)
Comment (nivista) [reply store/client.go:75]: Ok

**Pull request closed without merging**